src/main/java/org/folio/marc4ld/service/marc2ld/mapper/mapper/OtherVersionMapper.java

```diff
@@ -0,0 +1,183 @@
+    resource.setDoc(mapperHelper.getJsonNode(properties));
+    mapTitle(resource, workAndInstanceTitles.workTitle());
+    mapInstance(resource, workAndInstanceTitles.instanceTitle());
+    adjustWorkToInstanceRelation(resource);
+    resource.setLabel(workAndInstanceTitles.workTitle());
+  }
+
+  private WorkAndInstanceTitles getWorkAndInstanceTitles(MarcData marcData) {
+    String workTitle;
+    String instanceTitle;
+    var dataField = marcData.getDataField();
+    if (isSubfieldPresent(SUBFIELD_SRS_ID, dataField)) {
+      workTitle = getSubfieldValue(SUBFIELD_SRS_ID, dataField);
+      if (isSubfieldPresent(T, dataField)) {
+        instanceTitle = getSubfieldValue(T, dataField);
+      } else {
+        instanceTitle = getSubfieldValue(A, dataField) + " " + getSubfieldValue(B, dataField);
+      }
+    } else if (isSubfieldPresent(T, dataField)) {
+      return new WorkAndInstanceTitles(getSubfieldValue(T, dataField), getSubfieldValue(T, dataField));
+    } else {
+      return new WorkAndInstanceTitles(
+        getSubfieldValue(A, dataField) + " " + getSubfieldValue(B, dataField),
+        getSubfieldValue(A, dataField) + " " + getSubfieldValue(B, dataField)
+      );
+    }
+    return new WorkAndInstanceTitles(workTitle, instanceTitle);
+  }
+
+  private boolean isSubfieldPresent(char subfield, DataField dataField) {
+    return dataField.getSubfield(subfield) != null;
+  }
+
+  private String getSubfieldValue(char subfield, DataField dataField) {
+    return dataField.getSubfield(subfield).getData();
+  }
+
+  private void mapTitle(Resource resource, String title) {
+    getSubResourceOfPredicate(resource, TITLE)
+      .ifPresent(titleResource -> {
+        var properties = mapperHelper.getProperties(titleResource);
+        properties.put(MAIN_TITLE.getValue(), List.of(title));
+        titleResource.setDoc(mapperHelper.getJsonNode(properties));
+        labelService.setLabel(titleResource, properties);
+        titleResource.setId(hashService.hash(titleResource));
+      });
+  }
+
+  private static Optional<Resource> getSubResourceOfPredicate(Resource resource, PredicateDictionary predicate) {
+    return resource.getOutgoingEdges()
+      .stream()
+      .filter(re -> predicate.equals(re.getPredicate()))
+      .findFirst()
+      .map(ResourceEdge::getTarget);
+  }
+
+  private void mapInstance(Resource resource, String title) {
+    getSubResourceOfPredicate(resource, INSTANTIATES)
+      .ifPresent(instanceResource -> {
+        mapTitle(instanceResource, title);
+        instanceResource.setLabel(title);
+        setLabelsToLocalIds(instanceResource);
+        adjustIdentifiers(instanceResource);
+        instanceResource.setId(hashService.hash(instanceResource));
+      });
+  }
+
+  private void setLabelsToLocalIds(Resource resource) {
+    getIdentifiersOfTypes(resource, LOCAL_ID_TYPES)
+      .forEach(localIdResource -> {
+        var properties = mapperHelper.getProperties(localIdResource);
```

Comment:
Let us use LabelService

Reply:
Thank you! It was needed because ```$o``` was initially mapped to ```NAME``` property. Now it's not needed anymore.